extension/encoding/skywalkingencodingextension/extension.go

```diff
@@ -0,0 +1,28 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package skywalkingencodingextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/skywalkingencodingextension"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type skywalkingExtension struct {
+	unmarshaler ptrace.Unmarshaler
+}
+
+func (e *skywalkingExtension) UnmarshalTraces(buf []byte) (ptrace.Traces, error) {
+	return e.unmarshaler.UnmarshalTraces(buf)
+}
+
+func (e *skywalkingExtension) Start(_ context.Context, _ component.Host) error {
+	e.unmarshaler = skywalkingProtobufTrace{}
```

Comment:
This feels like a race condition waiting to happen, is there any reason why this can't be static for the lifetime of the extension instead of being set on Start?

Reply:
nice catch, I update here.